services/property-svc/internal/property/queries/v1/get_properties_by_subject_type.go

```diff
@@ -11,14 +11,19 @@ import (
 	"property-svc/repos/property_repo"
 )
 
-type GetPropertiesBySubjectTypeHandler func(ctx context.Context, subjectType pb.SubjectType) ([]*models.PropertyWithConsistency, error)
+type GetPropertiesQueryHandler func(ctx context.Context, subjectType *pb.SubjectType) ([]*models.PropertyWithConsistency, error)
 
-func NewGetPropertiesBySubjectTypeQueryHandler() GetPropertiesBySubjectTypeHandler {
-	return func(ctx context.Context, subjectType pb.SubjectType) ([]*models.PropertyWithConsistency, error) {
+func NewGetPropertiesQueryHandler() GetPropertiesQueryHandler {
+	return func(ctx context.Context, subjectType *pb.SubjectType) ([]*models.PropertyWithConsistency, error) {
 		propertyRepo := property_repo.New(hwdb.GetDB())
 
+		var subjectTypeID *int32
+		if subjectType != nil {
+			subjectTypeID = hwutil.PtrTo(int32(*subjectType))
+		}
```

Comment:
Why not pass `subjectType` directly to `SubjectType`?

Reply:
We cant. `subjectType` is of type `*pb.SubjectType` and `SubjectType` (I assume you mean the parameter for the query in l.25) needs the type *int32.